docs/modules/third_party_rules_callbacks.md

```diff
@@ -148,6 +148,49 @@ deny an incoming event, see [`check_event_for_spam`](spam_checker_callbacks.md#c
 
 If multiple modules implement this callback, Synapse runs them all in order.
 
+### `check_can_shutdown_room`
+
+_First introduced in Synapse v1.5X.0_
+
+```python
+async def check_can_shutdown_room(
+    user_id: str
+    room_id: str,
+) -> bool:
+```
+
+Called when an admin user requests the shutdown of a room. The module must return a
+boolean indicating whether the shutdown can go through. If the callback returns `False`,
+the shutdown will not proceed and the caller will see a `M_FOBIDDEN` error.
+
+If multiple modules implement this callback, they will be considered in order. If a
+callback returns `True`, Synapse falls through to the next one. The value of the first
+callback that does not return `True` will be used. If this happens, Synapse will not call
+any of the subsequent implementations of this callback.
+
+### `check_can_deactivate_user`
+
+_First introduced in Synapse v1.5X.0_
+
+```python
+async def check_can_deactivate_user(
+    requester: "synapse.types.Requester",
```

Comment:
Thinking about this a bit more, is there a reason we need to pass the `Requester` down to the module instead of just the user ID? If not I'd prefer if we passed the user ID for consistency (I know `on_create_room` already passes a `Requester` to modules, we'll probably need to fix it later).

Reply:
Oh, it was mostly so you could tell if the requester was an admin or not by inspecting the object. I can do an `is_admin` flag (or maybe there is a Module API to determine this) but I was going for convenience.

Reply:
Removed Requester, second parameter is now either the admin's user id, or None if no admin made the request.

Reply:
Having thought about it...I can't think of a reason to care who the admin is... going to use `by_admin`